ppa/archive/gale.py

```diff
@@ -0,0 +1,122 @@
+import logging
+import time
+
+import requests
+
+from django.conf import settings
+
+
+from ppa import __version__ as ppa_version
+
+
+logger = logging.getLogger(__name__)
+
+
+class GaleAPIError(Exception):
+    '''Base exception class for Gale API errors'''
+
+
+class GaleItemForbidden(GaleAPIError):
+    '''Permission denied to access item in Gale API'''
+
+
+class GaleItemNotFound(GaleAPIError):
+    '''Item not found in Gale API'''
+
+
+class GaleAPI:
+
+    #: base URL for all API requests
+    api_root = "https://api.gale.com/api"
+
+    instance = None
+
+    def __new__(cls):
+        # implement as a singleton
+        # adapted from https://softwareengineering.stackexchange.com/a/333710
+
+        # if no instance has been initialized, create and store on the class
+        if cls.instance is None:
+            cls.instance = super().__new__(cls)
+        # return the instance
+        return cls.instance
```

Comment:
love this choice! TIL that you can do it with `__new__`, which I didn't know existed.